packages/picker/src/Picker.ts

```diff
@@ -421,6 +425,15 @@ export class PickerBase extends SizedMixin(Focusable) {
         await this.menuStatePromise;
     }
 
+    public connectedCallback(): void {
+        if (!this.open) {
+            this.menuItems = [
+                ...this.querySelectorAll(`sp-menu-item`),
+            ] as MenuItem[];
```

Comment:
My thinking here is that it should be okay to change the menu items, but that it works best when the menu is closed.